apps/dc_tools/odc/apps/dc_tools/index_from_tar.py

```diff
@@ -1,13 +1,143 @@
 """ Index datasets from tar arachive
 """
 
+import os
 import sys
+from copy import deepcopy
+
 import click
 import datacube
 from datacube.utils.changes import allow_any
+from odc.index import from_yaml_doc_stream, odc_uuid
 from odc.io.tar import tar_doc_stream, tar_mode
 from odc.io.timer import RateEstimator
-from odc.index import from_yaml_doc_stream
+from pyproj import CRS, Transformer
+
+
+KNOWN_CONSTELLATIONS = [
+    'sentinel-2'
+]
+
+
+def _stac_product_lookup(item):
+    product_label = item['id']
+    product_name = item['properties']['platform']
+    region_code = None
+
+    constellation = item['properties']['constellation']
+
+    if constellation in KNOWN_CONSTELLATIONS:
+        if constellation == 'sentinel-2':
+            product_label = item['properties']["sentinel:product_id"]
```

Comment:
I usually prefer using `toolz.get_in` for nested access (`https://kite.com/python/docs/toolz.get_in`), since we do not do any schema validation we can't guarantee presence of various fields.

Mixing `'` and `"` on the same line is also not ideal.